lib/IRGen/GenCall.cpp

```diff
@@ -938,13 +938,16 @@ llvm::Type *SignatureExpansion::expandExternalSignatureTypes() {
         break;
       }
 
-      // If the coercion type is a struct, we need to expand it.
-      auto type = AI.getCoerceToType();
-      if (const auto *ST = dyn_cast<llvm::StructType>(type)) {
+      // If the coercion type is a struct which can be flattened, we need to
+      // expand it.
+      auto *ArgTy = AI.getCoerceToType();
+      if (AI.isDirect() && AI.getCanBeFlattened() &&
```

Comment:
IRGen uses lower-case convention: ArgTy -> argTy

AI.isDirect() is not necessary here. This code is guarded by  'case clang::CodeGen::ABIArgInfo::Direct'.

Reply:
Sure on the case thing, but the `AI.isDirect()` is needed as on L924, there is a `LLVM_FALLTHROUGH` for `clang::CodeGen::ABIArgInfo::Extend`.

Reply:
Oh, I missed that.